### ISSUE TITLE
Returning :reply from handle_event/3 in LiveComponent raises error

### ISSUE THREAD

**Issue by goldenprovider** [phoenixframework/phoenix_live_view#1207]
I am writing a LiveComponent with JS hook and I want to push event to that component in the JS hook, and get the reply from the component.
In JS hook (named TestHook):
```this.pushEventTo('#test-component', 'test_event', {}, (reply, ref) => {console.log(reply)})```
My component template:
```<div id="test-component" phx-hook="TestHook"></div>```
In component module:
```
def handle_event("test_event", _params, socket) do
  {:reply, %{result: "ok"}, socket}
end
```
This raises the error saying:
```
Expected: {:noreply, %Socket{}}
Got: {:reply, %{result: "ok"}, #Phoenix.LiveView.Socket<...>}
```

FYI: returning `:reply` from `handle_event/3` in LiveView works well.

My phoenix version: 1.5.4, phoenix_live_view version: 0.14.4

**Comment by chrismccord** [comment]
We need more information. What is your code? What is the stack trace? Please also fill out the issue template so we are aware of versions and such. Thanks!

**Comment by goldenprovider** [comment]
https://github.com/phoenixframework/phoenix_live_view/blob/78476dd7dfeb3038bed0b2d5b8dcaf8f0c607e19/lib/phoenix_live_view/channel.ex#L378

```
  defp inner_component_handle_event(component_socket, component, event, val) do
    :telemetry.span(
      [:phoenix, :live_component, :handle_event],
      %{socket: component_socket, component: component, event: event, params: val},
      fn ->
        case component.handle_event(event, val, component_socket) do
          {:noreply, %Socket{redirected: redirected, assigns: assigns} = component_socket} ->
            {
              {component_socket, {redirected, assigns.flash}},
              %{socket: component_socket, component: component, event: event, params: val}
            }

          other ->
            raise ArgumentError, """
            invalid return from #{inspect(component)}.handle_event/3 callback.
            Expected: {:noreply, %Socket{}}
            Got: #{inspect(other)}
            """
        end
      end
    )
  end
```
This method throws error finally.

**Comment by chrismccord** [comment]
Please open a new issue with your *full stack trace* and the issue template filled out. Thanks!

**Comment by Ziinc** [comment]
For anyone visiting this issue, it has been fixed in #1335 , currently pending release.
